### ISSUE TITLE
[bitnamit/etcd] Member id changes upon pod restart in 8.5.5

### ISSUE THREAD

**Issue by hasethuraman** [bitnami/charts#12832]
### Name and Version

bitnami/etcd 8.5.5

### What steps will reproduce the bug?

1. Run a 3 etcd replicas using 8.5.5 chart
2. Just delete the etcd-2 pod

The logs of the new etcd-2 pod instance is
```
	etcd 09:25:23.19 WARN  ==> You set the environment variable ALLOW_NONE_AUTHENTICATION=yes. For safety reasons, do not use this flag in a production environment.
	etcd 09:25:23.19 INFO  ==> Initializing etcd
	etcd 09:25:23.19 INFO  ==> Generating etcd config file using env variables
	etcd 09:25:23.31 INFO  ==> Detected data from previous deployments
	etcd 09:25:24.07 INFO  ==> Adding new member to existing cluster
	etcd 09:25:24.39 INFO  ==> Obtaining cluster member ID
	etcd 09:25:24.39 INFO  ==> Starting etcd in background
        ...
        Due to liveness probe etcd-2 restarts, this time the etcd logs just stops reporting the member is permanent removed and always stay in crashloop back.
```
The member list shows new member id for etcd-2 since prestop hook removed the older member

### What is the expected behavior?

The new etcd-2 must come up 

### What do you see instead?

etcd-2 pod stays in crashloop

### Additional information

I am trying to validate this behavior like the user who can scale-in and then later scale-out.

**Comment by doge95** [comment]
Encountering the same issue...

**Comment by bizonek27** [comment]
same issue since 8.3.8

**Comment by fmulero** [comment]
Hi @hasethuraman 

Have you set `removeMemberOnContainerTermination` to `false`? That will prevent the member removal and the new pod will use the same member ID as the previous one

**Comment by bizonek27** [comment]
Please check my workaround 
https://github.com/bitnami/charts/issues/12881

**Comment by bizonek27** [comment]
@hasethuraman 
I think removeMemberOnContainerTermination will disable automatic scaling down.

**Comment by fmulero** [comment]
If we set `removeMemberOnContainerTermination=false` I think the scale-up and and scale-down could cause problems because the new members are not removed.

I see several issues here, for instance, if `removeMemberOnContainerTermination=true` and one is restarted it couldn't start because it will try to use the same `member-id` and it was removed permanently. We can avoid storing the member-id in the persistence storage but in that case the data should be removed also. I need a further investigation

**Comment by tc1148** [comment]
We have run into this problem as well. I work for a very large corporation and we have been evaluating bitnami for etcd. This bug is a show-stopper. We cannot use bitnami with this issue. There appears to have been no progress in the last 3 months. Can someone provide an update? Thanks!

**Comment by fmulero** [comment]
We are working on it. I hope to be able to give you news soon

**Comment by corico44** [comment]
Hello @hasethuraman, @bizonek27, @doge95 and @tc1148,

Thank you very much for reporting this issue. The necessary changes to the etcd logic have been applied in the latest released version of etcd which is version **8.7.4**. So there should be no problem when removing a member having more than one replica and having the `removeMemberOnContainerTermination `parameter set to true.
